labelbox/data/serialization/ndjson/label.py

```diff
@@ -88,6 +88,39 @@ def _get_consecutive_frames(
             consecutive.append((group[0], group[-1]))
         return consecutive
 
+    @classmethod
+    def _get_segment_frame_ranges(
+        cls, annotation_group: List[Union[VideoClassificationAnnotation,
+                                          VideoObjectAnnotation]]
+    ) -> List[Tuple[int, int]]:
+        sorted_frame_segment_indices = sorted([
+            (annotation.frame, annotation.segment_index)
+            for annotation in annotation_group
+            if annotation.segment_index is not None
+        ])
+        if len(sorted_frame_segment_indices) == 0:
+            # Group segment by consecutive frames, since `segment_index` is not present
+            return cls._get_consecutive_frames(
+                sorted([annotation.frame for annotation in annotation_group]))
+        elif len(sorted_frame_segment_indices) == len(annotation_group):
```

Comment:
it's been a while since i have reviewed `annotation_group`, could you explain why this check is `== len(annotation_group)` ? Is it because we expect the number of segments to be the number of items in the annotation group?

Reply:
Yeah! So this is just checking that every annotation inside `annotation_group` contains a `segment_index`.
```
segment_index must be present in every VideoObjectAnnotation or VideoClassificationAnnotation. If only some annotations contain segment_index, error is raised
```